test/powershell/engine/Module/TestModuleManifest.Tests.ps1

```diff
@@ -217,3 +217,37 @@ Describe "Tests for circular references in required modules" -tags "CI" {
         { TestImportModule $false $false $true } | ShouldBeErrorId "Modules_InvalidManifest,Microsoft.PowerShell.Commands.ImportModuleCommand"
     }
 }
+
+Describe "Test-ModuleManifest Performance bug followup" -tags "CI" {
+
+$TestModulesFolder= 'testmodulerunspace'
+$TestModulesPath = Join-path $PSScriptRoot $TestModulesFolder
+$script:UserModulesPath = Join-path [System.AppDomain]::CurrentDomain.BaseDirectory '\Modules'
+
+
+# Install the Test Module
+Copy-Item $TestModulesPath\* $UserModulesPath -Recurse -Force
+
+    It "Test-ModuleManifest should not load unnessary modules" {
+
+        try 
+        {
+            $job = start-job -name "job1" -ScriptBlock {test-modulemanifest "$using:UserModulesPath\ModuleWithDependencies2\2.0\ModuleWithDependencies2.psd1" -verbose}
+
+        }
+        catch
+        {
+            throw $_
+        }
+
+        $verbose = $job.ChildJobs[0].Verbose.ReadAll()
+        $verbose.Count -le 50 | should be true
```

Comment:
Prior to the fix test-modulemanifest will load all the modules in the $pshome, creating a far more big verbose list. 

50 is a random small number that small enough to make sure the issue doesn't repro.

It can't be fixed as well since the verbose message contains other information that may vary.

I've changed the number to an even smaller number 20.